### ISSUE TITLE
Issues with some MIDIs starting

### ISSUE THREAD

**Issue by samboy** [samboy/ObHack#15]
If I play an entire map, then hit the exit and move on to the next map, sometimes the MIDI (song) for the new map does not start.  Every time this happens, I am making a transition from an older FreeDoom song to a newer FreeDoom song.  In particular, it happens when:

* Going from `MAP21` (`D_COUNT2`) to `MAP22` (`D_DDTBL3`)
* Going from `MAP24` (`D_THEDA3`) to `MAP25` (`D_ADRIAN`)
* Going from `MAP27` (`D_ROMER2`) to `MAP28` (`D_TENSE`)

I am using Doomsday 1.8.6.

The *workaround* is to save the game, warp to another level, then load the game; the music will then play for the entire level.

**Comment by samboy** [comment]
`d_count2` is a MIDI file, so the issue is not a MUS → MIDI transition.  

Let me see if I can reproduce it.

**Comment by samboy** [comment]
I can reliably reproduce the issue with the `MAP25` music reproducing using the default instant-exit levels and starting around level 19.  That in mind, I can start trying things like making *all* .mid files `.mus` files.

See also: https://doomwiki.org/wiki/MUS

**Comment by samboy** [comment]
Converting the newer Freedoom `.mid` files in to `.mus` files appears to solve the problem; I can no longer reproduce the issue after making the conversion.  In addition, the converted `.mus` files seamlessly loop (there’s a pause with the `.mid` form of the same songs) and sound like they play without issue after being converted.

The `MAP09` (`D_IN_CIT`) and MAP21 (`D_COUNT2`) files need to remain `.mid` files because the files get corrupted when converted in to `.mus` files.  There appears to be no issue starting these music files, so I will make an updated `DOOM2.WAD` and close this issue.

I have also added the tool to convert `.mid` files in to `.mus` files to the Git tree.

**Comment by samboy** [comment]
OK, I am in the process of releasing `ObHack-FreeDoom-3.3.9` with the `mid` → `mus` fix for this issue. Closing this ticket.

**Comment by samboy** [comment]
I have had to release ObHack-FreeDoom-3.4.0, which reverts `MAP11` (`D_STLKS2`), `MAP14` (`D_DDTBL2`), `MAP20` (`D_MESSAG`), and `MAP23` (`D_AMPIE`) to MIDI format; these are all files which got corrupted when converted from MIDI to MUS, but they play without problem as MIDIs in Doomsday 1.8.6.  I believe these MIDIs which can not cleanly become MUS files have been removed from FreeDoom proper because, while good music, they are buggy and there’s, here in 2020, no lack of open licensed MIDI files which we can use in FreeDoom (A paucity of open-licensed MIDIs *was* an issue back in 2015 when I last revised the music here).

With Doomsday 1.8.6, some files play best as MIDIs.  Other play better as MUS instead of MIDI files.  So, I have had to convert some but not all music files to get them to be as bug-free as possible in Doomsday 1.8.6.  

Annoying, and hopefully this will be the very last time I have to deal with music issues with my personal fork of FreeDoom.
